### DISCUSSION TITLE
Request for Enhanced Node Connectivity in Workflow Arrangement

### DISCUSSION THREAD

**Discussion by qwe4815124** [langgenius/dify#3632] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

The current system only allows for a maximum of two nodes to be connected to a single node. This restriction is illustrated in the attached screenshot, where you can see that no more than two nodes can flow into a single point.
This limitation hinders the ability to model more complex workflows that require multiple inputs converging into a single process node. For workflows that involve gathering multiple data streams or conditional paths merging, the two-node limit necessitates additional steps or alternative solutions that can complicate the overall design.
<img width="922" alt="image" src="https://github.com/langgenius/dify/assets/21254472/5e33a852-636c-4e29-9f7b-e01a489b1212">

### 2. Describe the feature you'd like to see

Could you please consider enhancing the software to support connecting more than two nodes to a single node? This feature would greatly improve the tool's capability to represent multifaceted workflows and reduce the need for workarounds.

### 3. How will this feature improve your workflow or experience?

I believe this enhancement would benefit many users who rely on dify for intricate workflow arrangements. Thank you for considering this request. I am looking forward to any updates you might have on this.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by crazywoola** [comment]
Maybe you mean things like this?

<img width="1424" alt="image" src="https://github.com/langgenius/dify/assets/100913391/55b615e4-c5a3-44a0-bbfe-58754cf1e73c">
<img width="1457" alt="image" src="https://github.com/langgenius/dify/assets/100913391/1a1f68d7-7437-40dd-865e-7f09c268a311">
